### ISSUE TITLE
[Help Wanted] Codebase accessibility refactor

### ISSUE THREAD

**Issue by oRazeD** [Pluglug/pie-menu-editor-fork#6]
There have already been several 1.18.7 refactor passes, but this issue goals to create more focus on the components of PME that should be refactored. This list is based on my personal standards for improving codebase accessibility and good coding practices. These changes would go a long way to making the codebase easier to interact with and much closer to PEP8 standards. None of the changes requested are required for shipping a new, stable version.

This list may evolve as expectations and understanding of the codebase changes:
- [x] Removal of all wildcard `*` imports
- [x] Remove of `import as` usage (Mostly cherry-picked imports)
- [x] Replacement of `bl_info` variable with Blender 4.2 `blender_manifest.toml` add-on registration standard
- [x] Add `_` to front of unused variables
- [x] Removal of Blender versioning code
- [ ] Removal of legacy compatibility fix functionality (?)
- [ ] Add [type-hinting](https://docs.python.org/3.9/library/typing.html) to, at minimum, all non-BPY classes and functions
- [ ] Removal of current debug, log, `print()`, and `traceback` implementations
    - [ ] Replace all references with simple [logging](https://docs.python.org/3/library/logging.html) implementation
    - [ ] Toggle via hotkey or User Preferences
- [ ] Replacement or removal of `eval()`  usage
- [ ] Replacement or removal of `exec()` usage
- [ ] Replace Python 2-era `%` / `%s` / `%d` string modification with f-strings or `str.format()`  for `logging`
- [ ] Test and replace bare `try` / `except` blocks with specific cases or alternative approach
- [ ] A few cases of accessing iterator elements post-loop execution https://github.com/Pluglug/pie-menu-editor-fork/blob/906b8ed09a5f89b98c84d819c204c41db197eb08/operators.py#L2690
- [ ] `tp` is not defined https://github.com/Pluglug/pie-menu-editor-fork/blob/a940aab27fd41b4dc45325b38356647f7d0dd706/ui.py#L43
- [ ] Many cases of comparing against callable (missing parentheses?) https://github.com/Pluglug/pie-menu-editor-fork/blob/a940aab27fd41b4dc45325b38356647f7d0dd706/preferences.py#L2322

**Comment by oRazeD** [comment]
@Pluglug Do you think you have anything to add to this Issue? I'm unsure if you can edit my tasks directly, I can transfer it to you if not.

**Comment by Pluglug** [comment]
Yes, `debug_utils` was the very first part of the PME modules that I managed to understand. I’ve even created some fan art related to it. [bcp - debug_utils.py](https://github.com/Pluglug/bone_color_presets/blob/main/debug_utils.py)
So, I think it’ll be a lot of fun to remove it!

I’m sorry, though—I misunderstood a bit and ended up clicking on a button out of curiosity. The issues I created can’t be undone, can they...? Were you intending to list out problems and handle them in stages?

**Comment by oRazeD** [comment]
> I’m sorry, though—I misunderstood a bit and ended up clicking on a button out of curiosity. The issues I created can’t be undone, can they...? Were you intending to list out problems and handle them in stages?

That's alright! That functionality is cool, I think it would be useful if we had a lot of issues to manage, but currently a single task with sub-tasks is probably good enough for this operation. In the future, if we feel like a specific bullet point deserves its own write-up, we can split off into new tasks as needed. I believe only admins can delete tasks.
